Rework OpenTelemetry logs page (#17269)
newrelic/docs-website: src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/best-practices/opentelemetry-best-practices-logs.mdx

---
title: OpenTelemetry logs in New Relic
tags:
  - Integrations
  - Open source telemetry integrations
  - OpenTelemetry
  - Logs
translate:
  - kr
metaDescription: Details on how New Relic works with OpenTelemetry logs
freshnessValidatedDate: 2024-05-17
redirects:
 - /docs/more-integrations/open-source-telemetry-integrations/opentelemetry/view-your-data/opentelemetry-logs-page
---

New Relic supports two primary workflows for ingesting logs generated with OpenTelemetry:

* [Direct to collector](https://opentelemetry.io/docs/specs/otel/logs/#direct-to-collector): 

>     * Applications can send logs directly to the dedicated New Relic OpenTelemetry Log Protocol (OTLP) endpoint. This streamlined approach minimizes additional components.

Author: ```suggestion
    * Applications can send logs directly to the New Relic OTLP endpoint.
```
OTLP != OpenTelemetry Log Protocol endpoint. "This streamlined approach minimizes additional components" repeats content on OpenTelemetry's docs. I think as a principle we need to avoid this because: 1. Less is more. 2. We're not the authority and have a history of getting it wrong. 3. The upstream doc can change and we might be left with a conflicting message.

Reviewer: @jack-berg What do you mean when you say "OTLP != OpenTelemetry Log Protocol endpoint"? How was my version (New Relic OpenTelemetry Log Protocol (OTLP) endpoint) incorrect? 

And re: your other point, I added that because it might be helpful to know when/why you'd use one workflow over the other, but we can keep the content more scrubbed down if we're depending on readers going to the OTEL docs.

Author: OTLP is an acronym for OpenTelemetry Protocol, not OpenTelemetry Log Protocol. It was originally an acronym for OpenTelemetry Line Protocol, but "Line" was later dropped after the shorted OTLP had already caught on. OTLP is a general purpose protocol for sending logs, metrics, traces, and profiles.

> And re: your other point, I added that because it might be helpful to know when/why you'd use one workflow over the other, but we can keep the content more scrubbed down if we're depending on readers going to the OTEL docs.

Yeah I get it. Its a balance - additional words can add helpful context even if they duplicate content which exists somewhere else, but also add more surface area for mistakes and content drift from the original source. I think in this case, the principle is to focus on "what" is happening with the two workflows, not "why".

    * The OpenTelemetry collector is typically used for this task. Scraped logs are then forwarded to the New Relic OTLP endpoint.
    * In-depth configuration information can be found in the following OpenTelemetry resources:
        * [OpenTelemetry collector](https://opentelemetry.io/docs/collector/)
        * [Filelog receiver with kubernetes](https://opentelemetry.io/docs/kubernetes/collector/components/#filelog-receiver)
        * [A practical example demonstrating this workflow for Java applications using Log4j2](https://github.com/newrelic/newrelic-opentelemetry-examples/tree/main/other-examples/java/logs-in-context-log4j2)

This documentation focuses on how New Relic processes OpenTelemetry logs received through its dedicated OTLP endpoint. Regardless of the chosen collection method, successful integration requires configuring your log source to export logs to this endpoint. Make sure to review the [endpoint configuration requirements](/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/best-practices/opentelemetry-otlp/#configure-endpoint-port-protocol) before proceeding.

> For information on configuring services with OpenTelemetry logs, see [OpenTelemetry APM monitoring](/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/get-started/apm-monitoring/opentelemetry-apm-intro/).

Reviewer: @jack-berg Let's clarify why they would want to go to the APM intro doc before continuing on this page. Are we saying they should already have APM OTEL monitoring set up, and if not, go see [OpenTelemetry APM monitoring](/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/get-started/apm-monitoring/opentelemetry-apm-intro/)?

Author: Yeah good point. I think this link should actually go under the "Direct to collector" bullet point. I want to tell readers that if they want to setup a service with APM log monitoring, this isn't the right page but go look at this other page.

Reviewer: @jack-berg Do you mean as a sub-bullet to "direct to collector", like this:

```suggestion
* [Direct to collector](https://opentelemetry.io/docs/specs/otel/logs/#direct-to-collector):
    * Applications can send logs directly to the New Relic OTLP endpoint. Refer to the relevant [OpenTelemetry language documentation](https://opentelemetry.io/docs/languages/) for specific implementation details.
    * For information on configuring services with OpenTelemetry logs, see [OpenTelemetry APM monitoring](/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/get-started/apm-monitoring/opentelemetry-apm-intro/).
```

Author: I meant just under the existing "Direct to collector -> Refer to the relevant OpenTelemetry language documentation" bullet. I pushed a commit to reflect this. WDYT?

> For information on configuring services with OpenTelemetry logs, see [OpenTelemetry APM monitoring](/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/get-started/apm-monitoring/opentelemetry-apm-intro/).

Reviewer: @jack-berg I think we need more info/preface for this link. Why would they want more info on configuring services with OTEL logs? Do they need to read that doc before returning to this one? More context would be helpful here

## OTLP log record mapping [#otlp-log-mapping]

> New Relic maps OTLP log records to the `Log` data type. The table below describes how fields from the [`LogRecord` proto message](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto) are interpreted:

Reviewer: ```suggestion
New Relic maps OTLP log records to the `Log` data type. The table below describes how fields from the [`LogRecord` proto message](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto) map to New Relic `logs`:
```

> <table>

Author: The HTML encoding of this table is much harder to read than markdown. Is there some sort of reason or policy behind this?

---

> title: OpenTelemetry logs in New Relic

Reviewer: @jack-berg Since this page (along with metrics, resources, and traces pages) are no longer "best practices", I think we should upate the file path names. Perhaps something like `/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/data/logs`. Thoughts?

Author: I agree. Was holding off to reduce churn in case we revisit the navigation structure again. Why don't we see where things settle in the next week or two, and if we're comfortable with the navigation, update the path names to reflect it?

translate:
  - kr
metaDescription: Details on how New Relic works with OpenTelemetry logs
freshnessValidatedDate: 2024-05-17
redirects:
 - /docs/more-integrations/open-source-telemetry-integrations/opentelemetry/view-your-data/opentelemetry-logs-page
---

> New Relic supports two primary workflows for ingesting logs generated with OpenTelemetry:

Author: This phrasing isn't correct. These aren't new relic's workflows and we don't explicitly support them. These are the workflows identified / documented by the OpenTelemetry project. At the end of the day, New Relic supports OTLP ingest.

Reviewer: @jack-berg I'm open to changing the phrasing, but I want to keep it New Relic-focused and communicate that we "support" the two OTLP ingest workflows.

How about something like "To send OTLP-ingested logs to New Relic, use one of the following OpenTelemetry workflows:"

I'm open to suggestions.
